server/integration/infinispan/src/main/java/org/jboss/as/clustering/infinispan/subsystem/InfinispanSubsystemXMLReader.java

```diff
@@ -349,15 +358,190 @@ private void parseContainer(XMLExtendedStreamReader reader, PathAddress subsyste
                         break;
                     }
                 }
+                case COUNTERS: {
+                   if (namespace.since(Namespace.INFINISPAN_SERVER_9_2)) {
+                       this.parseCounters(reader, countersConfigurationAddress, operations);
+                       break;
+                   }
+                }
+                default: {
+                    throw ParseUtils.unexpectedElement(reader);
+                }
+            }
+        }
+    }
+
+    private void parseCounters(XMLExtendedStreamReader reader, PathAddress countersConfigurationAddress,
+            Map<PathAddress, ModelNode> operations) throws XMLStreamException {
+
+        for (int i = 0; i < reader.getAttributeCount(); i++) {
+            String value = reader.getAttributeValue(i);
+            Attribute attribute = Attribute.forName(reader.getAttributeLocalName(i));
+            switch (attribute) {
+                case NUM_OWNERS: {
+                    //TODO
+                    break;
+                }
+                case RELIABILITY: {
+                    //TODO
+                    break;
+                }
+                default: {
+                    throw ParseUtils.unexpectedAttribute(reader, i);
+                }
+            }
+        }
 
+        while (reader.hasNext() && (reader.nextTag() != XMLStreamConstants.END_ELEMENT)) {
+            Element element = Element.forName(reader.getLocalName());
+            switch (element) {
+                case STRONG_COUNTER: {
+                    parseStrongCounterElement(reader, element, countersConfigurationAddress, operations);
+                    break;
+                }
+                case WEAK_COUNTER: {
+                    parseWeakCounterElement(reader, element, countersConfigurationAddress, operations);
+                    break;
+                }
                 default: {
                     throw ParseUtils.unexpectedElement(reader);
                 }
             }
         }
     }
 
-    private void parseGlobalState(XMLExtendedStreamReader reader, PathAddress containerAddress,
+    private void parseStrongCounterElement(XMLExtendedStreamReader reader, Element element,
+            PathAddress countersConfigurationAddress, Map<PathAddress, ModelNode> operations)
+            throws XMLStreamException {
+
+        PathAddress strongCounterAddress = countersConfigurationAddress;
+        ModelNode counter = Util.createAddOperation(strongCounterAddress);
+        for (int i = 0; i < reader.getAttributeCount(); i++) {
+            String value = reader.getAttributeValue(i);
+            Attribute attribute = Attribute.forName(reader.getAttributeLocalName(i));
+            switch (attribute) {
+                case INITIAL_VALUE: {
+                    ((SimpleAttributeDefinition) StrongCounterConfigurationResource.INITIAL_VALUE)
```

Comment:
If you declared these as a `SimpleAttributeDefinition` in `CounterConfigurationResource` there is no need to cast

Reply:
We should really remove all of these casts in this file as they aren't necessary if ^ action is taken